src/julia-parser.scm

```diff
@@ -1141,6 +1141,14 @@
     (begin0 (list 'type (if mut? 'true 'false) sig (parse-block s))
             (expect-end s word))))
 
+;; consume any number of line endings from a token stream
+(define (take-lineendings s)
+  (let ((nt (peek-token s)))
+        (if (or (newline? nt) (eqv? nt #\;))
+            (begin (take-token s)
+                   (take-lineendings s))
+            s)))
```

Comment:
Try to find a good scheme mode for your editor. This should look like

```
(define (take-lineendings s)
  (let ((nt (peek-token s)))
    (if (or (newline? nt) (eqv? nt #\;))
        (begin (take-token s)
               (take-lineendings s))
        s)))
```

(The `if` was indented to where `let` variables would go.)